packages/api-v2/src/plan/plan.service.ts

```diff
@@ -201,8 +202,30 @@ export class PlanService {
           this.formatPlanServiceCtx("update")
         );
 
+        throw new InvalidConcentration();
+        return null;
+      }
+
+      const isValidMajorCatalogueYear = this.majorService.isValidCatalogueYear(
```

Comment:
Ok so I think this mostly makes sense to me. However I think the order of these needs to be switched up a little. Right now, we are checking if the concentration is valid, and then we check if the major is valid for the given catalogue year. However, we should really be checking if we support the major in the given year FIRST, and then we should check if the concentration exist for that particular major and year. Let me know if you agree/disagree with my logic because I may be wrong, but I think this seems the most correct to me